src/schemas/pa-yaml/v3.0/pa.schema.yaml

```diff
@@ -435,24 +435,42 @@ definitions:
   DataSource-name:
     $ref: "#/definitions/entity-name"
 
-  DataSource-Type:
-    enum:
-      - DataverseTable
-
   DataSource-instance:
     type: object
-    required: [Type]
+    required:
+      - Type
     properties:
-      Type: { $ref: "#/definitions/DataSource-Type" }
+      Type:
+        type: string
+        enum:
+          - Table
+          - Actions
+      ConnectorId:
+        type: string
+        nullable: true
+      Parameters:
+        type: object
+        additionalProperties:
+          type: string
+    additionalProperties: false
     oneOf:
       - required: [Type]
         additionalProperties: false
         properties:
           Type:
-            const: DataverseTable
-          TableLogicalName:
-            type: string
-            minLength: 1
+            const: Table
+          Parameters:
+            type: object
+            properties:
+              TableLogicalName:
+                type: string
+            additionalProperties: false
```

Comment:
nit: I put the 'additionalProperties' before the proeprties since the properties is often quite long on some types. just makes it easier to find and consistent with the rest of the file